eZ/Publish/Core/Persistence/Legacy/Content/Type/Gateway/DoctrineDatabase.php

```diff
@@ -1540,4 +1540,57 @@ public function removeFieldDefinitionTranslation(
 
         $deleteQuery->execute();
     }
+
+    /**
+     * Removes types created or modified by the user.
+     */
+    public function removeByUserAndVersion(int $userId, int $version): void
+    {
+        $this->connection->beginTransaction();
+
+        $qb = $this->connection->createQueryBuilder();
+        $qb->delete('ezcontentclass')
+            ->where('creator_id = :user or modifier_id = :user')
+            ->andWhere('version = :version')
+            ->setParameter('user', $userId, ParameterType::INTEGER)
+            ->setParameter('version', $version, ParameterType::INTEGER);
+
+        $qb->execute();
+
+        $this->cleanupAssociations();
+
+        $this->connection->commit();
+    }
+
+    private function cleanupAssociations()
+    {
+        $this->cleanupClassAttributeTabel();
+        $this->cleanupClassAttributeMLTabel();
+        $this->cleanupClassGroupTabel();
+        $this->cleanupClassNameTabel();
+    }
+
+    private function cleanupClassAttributeTabel()
```

Comment:
Typo: "Table". Moreover there's no point in saying that's it's about table. It's rather what GW's are meant to operate on.
```suggestion
    private function cleanupClassAttributes(): void
```

Reply:
Those functions are private and are absent in the interface.

Reply:
Also, those are internal operations in doctrine implementation.

Reply:
> Those functions are private and are absent in the interface.
> Also, those are internal operations in doctrine implementation.

That is really beside the point...

Reply:
I will fix the typo. Sufix `Table` in my opinion is important on this level of abstraction.